### PR TITLE
Add cn.feishu.Feishu

### PR DESCRIPTION
**Pull request by chenyuanrun**
Feishu, the advanced enterprise collaboration and management platform - integrating instant communication, calendar, audio/video conferencing, cloud documents, cloud disk, workbench and other functions - accomplishes organizations and individuals, more efficiently and more pleasantly.

- Home page: https://www.feishu.cn/
- Linux download page: https://www.feishu.cn/download

### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by chenyuanrun** [comment]
bot, build

**Comment by chenyuanrun** [comment]
bot, build cn.feishu.Feishu

File: cn.feishu.Feishu.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>cn.feishu.Feishu</id>
+  <update_contact>contact@feishu.cn</update_contact>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Feishu</name>
+  <summary>飞书(Feishu)</summary>
```
Comment (bbhtt) [inline cn.feishu.Feishu.metainfo.xml:7]: THis is not an acceptable summary

File: cn.feishu.Feishu.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  <description>
+    <p>
+    飞书(Feishu)
```
Comment (bbhtt) [inline cn.feishu.Feishu.metainfo.xml:10]: Description should be descriptive not one word.

File: cn.feishu.Feishu.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <url type="homepage">https://feishu.cn/</url>
+  <developer_name>Feishu Linux Team</developer_name>
+  <releases>
+    <release version="6.9.20" date="2023-08-11">
+      <url type="details">https://www.feishu.cn/hc/zh-CN/articles/872140099131-v6.9-%E7%94%BB%E6%9D%BF%E4%BC%97%E5%A4%9A%E6%96%B0%E7%8E%A9%E6%B3%95%E6%9D%A5%E8%A2%AD-%E5%8A%A9%E5%8A%9B%E7%BB%98%E5%9B%BE%E4%BD%93%E9%AA%8C%E5%86%8D%E5%8D%87%E7%BA%A7</url>
+      <description>Feishu v6.9.20</description>
+    </release>
+  </releases>
+  <launchable type="desktop-id">cn.feishu.Feishu.desktop</launchable>
+  <content_rating type="oars-1.0">
```
Comment (bbhtt) [inline cn.feishu.Feishu.metainfo.xml:22]:
```suggestion
  <content_rating type="oars-1.1">
```

File: cn.feishu.Feishu.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  # - --socket=wayland
+  - --socket=pulseaudio
+  # For ibus
+  # - --socket=session-bus
+  # - --socket=system-bus
+  # For camera
+  - --device=all
+  # For ptrace
+  - --allow=devel
+  - --filesystem=home
```
Comment (bbhtt) [inline cn.feishu.Feishu.yaml:21]: why is this needed?

**Comment by bbhtt** [comment]
Please complete the checklist as above

File: cn.feishu.Feishu.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  - --socket=x11
+  # - --socket=wayland
+  - --socket=pulseaudio
+  # For ibus
+  # - --socket=session-bus
+  # - --socket=system-bus
+  # For camera
+  - --device=all
+  # For ptrace
+  - --allow=devel
```
Comment (hfiguiere) [inline cn.feishu.Feishu.yaml:20]: why do you need `devel`?

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+    "only-arches": ["x86_64"],
+    "automerge-flathubbot-prs": false
```
Comment (hfiguiere) [inline flathub.json:3]:
```suggestion
    "only-arches": ["x86_64"]
```

File: cn.feishu.Feishu.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        commands:
+          - ar x Feishu-linux.deb
+          - tar -xf data.tar.xz
+          - mv opt/bytedance/feishu .
+          - sed '/^exec -a/d' feishu/bytedance-feishu > feishu/bytedance-feishu.sh
+          - rm -rf opt usr *.tar.* debian-binary *.deb
+        dest-filename: apply_extra
+      - type: extra-data
+        filename: Feishu-linux.deb
+        url: https://sf3-cn.feishucdn.com/obj/ee-appcenter/6085d1c4/Feishu-linux_x64-6.9.20.deb
```
Comment (hfiguiere) [inline cn.feishu.Feishu.yaml:53]: is resdistribution EXPLICITLY allowed?

File: cn.feishu.Feishu.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        # url: http://127.0.0.1:8088/Feishu-linux.deb
```
Comment (hfiguiere) [inline cn.feishu.Feishu.yaml:54]:
```suggestion
```

File: cn.feishu.Feishu.yaml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+          - source /app/extra/feishu/bytedance-feishu.sh
+          - exec zypak-wrapper /app/extra/feishu/feishu "$@"
+        dest-filename: feishu
+      - type: script
+        commands:
+          - ar x Feishu-linux.deb
+          - tar -xf data.tar.xz
+          - mv opt/bytedance/feishu .
+          - sed '/^exec -a/d' feishu/bytedance-feishu > feishu/bytedance-feishu.sh
+          - rm -rf opt usr *.tar.* debian-binary *.deb
```
Comment (hfiguiere) [inline cn.feishu.Feishu.yaml:49]:
```suggestion
```

You don't need to remove anything in th build dir.

File: cn.feishu.Feishu.yaml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+command: feishu
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  # - --socket=wayland
+  - --socket=pulseaudio
+  # For ibus
+  # - --socket=session-bus
+  # - --socket=system-bus
```
Comment (hfiguiere) [inline cn.feishu.Feishu.yaml:16]:
```suggestion
```

**Comment by hfiguiere** [comment]
Also you missed:

> Please describe your application briefly.

File: cn.feishu.Feishu.yaml (hunk 7 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml:53]: No, so I have to install it via extra-data.

File: cn.feishu.Feishu.yaml (hunk 9 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml:49]: These are extracted from extra-data and will not be used anymore, I think it is better to remove those useless files to save disk.

File: cn.feishu.Feishu.metainfo.xml (hunk 1 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.metainfo.xml]: I have changed the summary.

File: cn.feishu.Feishu.metainfo.xml (hunk 2 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.metainfo.xml]: I have changed the description.

File: cn.feishu.Feishu.metainfo.xml (hunk 3 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.metainfo.xml]: I have changed this.

File: cn.feishu.Feishu.yaml (hunk 4 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml:19]: Feishu would call ptrace for some reason, I have disable it and feishu still run well.

**Comment by chenyuanrun** [comment]
> Please complete the checklist above

Done.

File: cn.feishu.Feishu.yaml (hunk 10 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml]: Done.

**Comment by chenyuanrun** [comment]
> Also you missed:
> 
> > Please describe your application briefly.

Done.

**Comment by chenyuanrun** [comment]
@bbhtt @hfiguiere May you review this PR again? THANKS!

**Comment by chenyuanrun** [comment]
bot, build cn.feishu.Feishu

File: cn.feishu.Feishu.yaml (hunk 4 above)
Comment (bbhtt) [reply cn.feishu.Feishu.yaml:19]: I meant why is home access needed?

File: cn.feishu.Feishu.yaml (hunk 5 above)
Comment (bbhtt) [reply cn.feishu.Feishu.yaml]: https://github.com/flathub/flathub/pull/5102#discussion_r1545556089

File: cn.feishu.Feishu.yaml (hunk 4 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml:19]: Feishu is a collaboration and management platform integrating instant communication, I think users can share files and docs with each other easier if feishu can access home dir.

File: cn.feishu.Feishu.yaml (hunk 9 above)
Comment (hfiguiere) [reply cn.feishu.Feishu.yaml:47]: nevermind I misread it's the apply_extra.

File: cn.feishu.Feishu.yaml (hunk 8 above)
Comment (chenyuanrun) [reply cn.feishu.Feishu.yaml]: Done
